src/test/java/uk/gov/mca/beacons/api/gateways/AuthGatewayImplTest.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.mca.beacons.api.gateways;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static uk.gov.mca.beacons.api.gateways.AuthGatewayImpl.SupportedPermissions;
+
+import com.azure.spring.aad.webapi.AADOAuth2AuthenticatedPrincipal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import uk.gov.mca.beacons.api.domain.User;
+
+@ExtendWith(MockitoExtension.class)
+class AuthGatewayImplTest {
+
+  private AuthGateway authGateway;
+
+  SimpleGrantedAuthority updaterAuth = new SimpleGrantedAuthority(
+    SupportedPermissions.APPROLE_UPDATE_RECORDS.toString()
+  );
+  SimpleGrantedAuthority unsupportedAuth1 = new SimpleGrantedAuthority(
+    "JUNK_AUTH_1"
+  );
+  SimpleGrantedAuthority unsupportedAuth2 = new SimpleGrantedAuthority(
+    "JUNK_AUTH_2"
+  );
+
+  Authentication authentication = mock(Authentication.class);
+  SecurityContext context = mock(SecurityContext.class);
+  AADOAuth2AuthenticatedPrincipal principal = mock(
+    AADOAuth2AuthenticatedPrincipal.class
+  );
+
+  @BeforeEach
+  void before() {
+    given(context.getAuthentication()).willReturn(authentication);
+    given(authentication.getPrincipal()).willReturn(principal);
+    SecurityContextHolder.setContext(context);
+    authGateway = new AuthGatewayImpl();
+  }
```

Comment:
Discussed on mid-sprint with @tm1303 the option to _not_ test I/O and mock things we don't own.  Option to only unit test the mapping logic at line ~25 in the class.  Will leave it up to you :-)